tests/reporters/unittest_reporting.py

```diff
@@ -334,6 +336,44 @@ def test_multi_format_output(tmp_path: Path) -> None:
     )
 
 
+def test_multi_reporter_independant_messages() -> None:
+    """Messages should not be modified by multiple reporters"""
+
+    check_message = "Not modified"
+
+    class ReporterModify(BaseReporter):
+        def handle_message(self, msg: Message) -> None:
+            msg.msg = "Modified message"
+
+        def writeln(self, string: str = "") -> None:
+            pass
+
+        def _display(self, layout: Section) -> None:
+            pass
+
+    class ReporterCheck(BaseReporter):
+        def handle_message(self, msg: Message) -> None:
+            assert msg.msg == check_message
+
+        def writeln(self, string: str = "") -> None:
+            pass
+
+        def _display(self, layout: Section) -> None:
+            pass
+
+    multi_reporter = MultiReporter([ReporterModify(), ReporterCheck()], lambda: None)
+
+    message = Message(
+        symbol="missing-docstring",
+        msg_id="C0123",
+        location=MessageLocationTuple("abspath", "path", "module", "obj", 1, 2, 1, 3),
+        msg=check_message,
+        confidence=HIGH,
+    )
+
+    multi_reporter.handle_message(message)
+
```

Comment:
It seems the assert is missing ? Something like this maybe:
```suggestion

assert multi_reporter.msg = check_message 
```

Reply:
I putted the assert in the  `CheckReporter` to ensure he was not receiving a modified msg. Maybe it is more clear if the check is outside?

Reply:
Ha I missed that. Yeah it's better outside 👍

Reply:
@Pierre-Sassoulas, I added the assert in the test but also keep the one with the reporter. This ensure each reporter is independant, but also that the handle_message in multi reporter does not change the message itself (if in the future this function is modified).

Reply:
Thank you !